clientconn.go

```diff
@@ -1740,8 +1744,13 @@ func (cc *ClientConn) parseTargetAndFindResolver() error {
 	// We are here because the user's dial target did not contain a scheme or
 	// specified an unregistered scheme. We should fallback to the default
 	// scheme, except when a custom dialer is specified in which case, we should
-	// always use passthrough scheme.
-	defScheme := resolver.GetDefaultScheme()
+	// always use passthrough scheme. For either case, we need to respect any overridden
+	// global defaults set by the user.
+	defScheme := resolver.GetDefaultSchemeOrDNS()
+	if cc.dopts.eagerConnect {
+		defScheme = resolver.GetDefaultSchemeOrPassthrough()
+	}
```

Comment:
Based on the above, this could be:

```go
defScheme := cc.dopts.defScheme
if defScheme == "" || internal.UserSetDefaultScheme {
	defScheme = resolver.GetDefaultScheme()
}
```